modules/ingester/instance_search_test.go

```diff
@@ -159,3 +160,84 @@ func TestInstanceSearchNoData(t *testing.T) {
 	assert.NoError(t, err)
 	require.Len(t, sr.Traces, 0)
 }
+
+func TestInstanceSearchDoesNotRace(t *testing.T) {
```

Comment:
This currently detects a race:
```
WARNING: DATA RACE
Write at 0x00c000111d40 by goroutine 470:
  runtime.mapassign_fast64()
      /usr/local/go/src/runtime/map_fast64.go:92 +0x0
  github.com/grafana/tempo/tempodb/encoding.(*appender).Append()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/tempodb/encoding/appender.go:60 +0x3cf
  github.com/grafana/tempo/modules/search.(*StreamingSearchBlock).Append()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/search/streaming_search_block.go:90 +0x5b3
  github.com/grafana/tempo/modules/ingester.(*instance).writeTraceToHeadBlock()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/ingester/instance.go:498 +0x26b
  github.com/grafana/tempo/modules/ingester.(*instance).CutCompleteTraces()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/ingester/instance.go:185 +0x2a4
  github.com/grafana/tempo/modules/ingester.TestInstanceSearchDoesNotRace.func3()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/ingester/instance_search_test.go:217 +0x5c
  github.com/grafana/tempo/modules/ingester.TestInstanceSearchDoesNotRace.func1()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/ingester/instance_search_test.go:190 +0x4b

Previous read at 0x00c000111d40 by goroutine 474:
  github.com/grafana/tempo/tempodb/encoding.(*appender).Records()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/tempodb/encoding/appender.go:67 +0x8a
  github.com/grafana/tempo/modules/search.(*StreamingSearchBlock).Search()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/search/streaming_search_block.go:96 +0x70
  github.com/grafana/tempo/modules/ingester.(*instance).searchWAL.func1()
      /Users/annanay/Desktop/git/go/src/github.com/grafana/tempo/modules/ingester/instance_search.go:98 +0xe6
```

Reply:
I think we discussed this one already @mdisibio 

Some searchBlock specific mutexes need to be set up I guess.